models/model.py

```diff
@@ -54,7 +60,16 @@ def __init__(self, model_hparams, optimizer_hparams):
         self.lr_scheduler = None
 
     def configure_optimizers(self):
-        optimizer = AdamW(self.parameters(), lr=self.hparams.optimizer_hparams['lr'],
+        params = list(self.named_parameters())
+
+        def is_encoder(n): return n.startswith('model')
```

Comment:
Does this also work for both GNN architectures?

Reply:
Curious about this indeed

Reply:
Good point. I haven't tested it with the new changes you made. But this part is generic. All the encoders (roberta, and both gnn) are stored in ```self.model```, so there layer name will start with "model".

Reply:
Yea it makes sense now, I thought 'model' might be some magic keyword in Roberta

Reply:
I think @RedHilarious actually renamed this file so if that impacts the model name, we should reverse that renaming as we already trained the models with from the models.py file

Reply:
The string "model" is linked to the named of the parameter ```self.model``` (if the parameter was named ```encoder```), then the string would need to be "encoder". In this PR (and in main) that's the correct name. What changes you are referring to @fvlieshout ?

Reply:
In the other PR from Rahel, she renamed model.py to document_classifier.py (or something similar). I somehow read that the naming of the python file would influence the names of the model layers haha. But nevermind, I just misread!